### PR TITLE
Memory saver: lazy loading of source classes

### PR DESCRIPTION
**Pull request by tarekziade**
## Closes  https://github.com/elastic/connectors-python/issues/684

Removes `get_source_klass_dict,` which loads all sources and imports all libs, even if they are never used.

This saves up to 40MiB of heap and counting

### CODE REVIEW

**Comment by tarekziade** [comment]
Impact on memory

- with the patch - https://buildkite.com/elastic/connectors-python-nightly/builds/472
- without the patch - https://buildkite.com/elastic/connectors-python-nightly/builds/469

File: connectors/services/job_scheduling.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
 from connectors.es.index import DocumentNotFoundError
 from connectors.logger import logger
 from connectors.services.base import BaseService
-from connectors.source import get_source_klass_dict
+from connectors.source import get_source_klass, get_source_list
```
Comment (wangch079) [inline connectors/services/job_scheduling.py:27]: Don't feel it's necessary to introduce these 2 methods, which just return dict keys and a dict.

Comment (artem-shelkovnikov) [reply connectors/services/job_scheduling.py:27]: Sorry I did not get your comment @wangch079, what do you mean?

Comment (tarekziade) [reply connectors/services/job_scheduling.py:27]: oh I see ! you're right :)

File: connectors/services/job_scheduling.py [hunk 2]
```diff
@@ -23,7 +23,9 @@
 )
 from connectors.logger import logger
 from connectors.services.base import BaseService
-from connectors.source import get_source_klass_dict
+from connectors.source import get_source_klass
+
+DEFAULT_MAX_CONCURRENT_SYNCS = 1
```
Comment (wangch079) [inline connectors/services/job_scheduling.py:28]: This is not needed here

Comment (tarekziade) [reply connectors/services/job_scheduling.py:28]: probably a merge flake, thanks

**Review by wangch079 — APPROVED** [review]

**Pull request merged**